logic/transactions/pool.js

```diff
@@ -0,0 +1,705 @@
+		return inList !== undefined;
+	}).length > 0;
+};
+
+/**
+ * Validates address.
+ * @param {string} address
+ * @return {null|err}
+ * @todo check address against mem_accounts
+ */
+__private.checkAddress = function (address) {
+	if (address.charAt(address.length-1) !== 'L') {
+		return 'Invalid address, last char must be "L"';
+	} 
+	if (address.slice(0,address.length-1).match(/^[0-9]+$/) === null) {
+		return 'Invalid address, must be numbers';
+	}
+	return null;
+};
+
+/**
+ * Validates publicKey.
+ * @param {string} publicKey
+ * @return {null|err}
+ * @todo check publicKey against mem_accounts
+ */
+__private.checkPublicKey = function (publicKey) {
+	if (typeof publicKey !== 'string') {
+		return 'Invalid public key, must be a string';
+	}
+	if (publicKey.length !== 64) {
+		return 'Invalid public key, must be 64 characters long';
+	}
+	if (publicKey.match(/^[0-9A-Fa-f]+$/) === null) {
+		return 'Invalid public key, must be a hex string';
+	}
+	return null;
+};
+
+/**
+ * Adds transactions to pool list.
+ * Checks if tx is in pool. Checks pool limit.
+ * @implements {__private.transactionInPool}
+ * @param {transaction} transaction
+ * @param {Object} poolList
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+__private.add = function (transaction, poolList, cb) {
+	if (__private.countTxsPool() >= self.poolStorageTxsLimit) {
+		return setImmediate(cb, 'Transaction pool is full');
+	}
+	if (__private.transactionInPool(transaction.id)) {
+		return setImmediate(cb, 'Transaction is already in pool: ' + transaction.id);
+	} else {
+		poolList.transactions.push(transaction);
+		var index = poolList.transactions.indexOf(transaction);
+		poolList.index[transaction.id] = index;
+		self.processed++;
+		return setImmediate(cb);
+	}
+};
+
+/**
+ * Adds transactions to pool list.
+ * Clear transaction if is in pool.
+ * @implements {__private.transactionInPool}
+ * @param {transaction} transaction
+ * @param {Object} poolList
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+__private.addReady = function (transaction, poolList, cb) {
+	self.delete(transaction);
+	poolList.transactions.push(transaction);
+	var index = poolList.transactions.indexOf(transaction);
+	poolList.index[transaction.id] = index;
+	self.processed++;
+	return setImmediate(cb);
+};
+
+/**
+ * Deletes id from pool list index.
+ * @param {string} id
+ * @param {Object} poolList
+ * @return {boolean} true if transaction id is on the list and was deleted
+ */
+__private.delete = function (id, poolList) {
+	var index = poolList.index[id];
+
+	if (index !== undefined) {
+		poolList.transactions[index] = false;
+		delete poolList.index[id];
+		return true;
+	}
+	return false;
+};
+
+/**
+ * Sums unverified, verified.pending and verified.ready indexes length.
+ * @return {Number} unverified, pending, ready
+ */
+__private.countTxsPool = function () {
+	var totalPool = self.getUsage();
```

Comment:
private function shouldn't call public function - it should be opposite, some logic is wrong here, also `getUsage` give us more data that we actually need - is not good in term of performance